pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/ClusterResources.java

```diff
@@ -48,5 +104,33 @@ public FailureDomainResources(MetadataStore store, Class<FailureDomainImpl> claz
                 int operationTimeoutSec) {
             super(store, clazz, operationTimeoutSec);
         }
+
+        public List<String> listFailureDomains(String clusterName) throws MetadataStoreException {
+            return getChildren(joinPath(BASE_CLUSTERS_PATH, clusterName, FAILURE_DOMAIN));
+        }
+
+        public Optional<FailureDomainImpl> getFailureDomain(String clusterName, String domainName)
+                throws MetadataStoreException {
+            return get(joinPath(BASE_CLUSTERS_PATH, clusterName, FAILURE_DOMAIN, domainName));
+        }
+
+        public void deleteFailureDomain(String clusterName, String domainName) throws MetadataStoreException {
+            delete(joinPath(BASE_CLUSTERS_PATH, clusterName, FAILURE_DOMAIN, domainName));
+        }
+
+        public void deleteFailureDomains(String clusterName) throws MetadataStoreException {
+            String failureDomainPath = joinPath(BASE_CLUSTERS_PATH, clusterName, FAILURE_DOMAIN);
+            for (String domain : getCache().getChildren(failureDomainPath).join()) {
```

Comment:
Is it correct to use 'cache' here?

Reply:
In this case it's redundant but correct. The exists and getChildren are cached directly in the MetadataStore implementation, the MetadataCache is for keeping a typed cache of the objects. 
The delete operation on the cache is just a convenience method that is just forwarded to the store implementation

Reply:
Makes sense. Thanks